LSPS0/README.md

```diff
@@ -0,0 +1,606 @@
+information in the optional `data` object if the error `code`
+defines it.
+If the client does not recognize the error `code`, or is
+expecting a field in the `data` object that is not present, it
+SHOULD indicate this as an "unrecognized" error to the user.
+
+> **Rationale** LSPs might write incorrect or misleading
+> human-readable error messages, and users might report such
+> error messages as bugs to the client developers, since the
+> user-visible source of the error message would be the client;
+> it is thus better if the client writes its own error
+> messages that it can change based on user feedback.
+> Newer versions of some LSPS spec may introduce new error codes,
+> or the specification may be incomplete and actual development
+> shows that some unspecified error could possibly occur, in
+> which case the human-readable `error` field could contain a
+> description of this error, which developers of clients can
+> then use to help guide the evolution of the specification, or
+> to comply with later versions of the LSPS spec.
+
+#### Disconnection Handling
+
+Networks are unreliable, and network participants are also unreliable.
+
+Clients MUST use a high-entropy `id` string for JSON-RPC 2.0 requests,
+such as a UUID, or a hex encoding of a random binary blob of at least
+80 bits, with randomness acquired from a cryptographically secure
+source.
+Clients MUST NOT use just a simple incrementing counter for the `id`.
+
+If a client receives a JSON-RPC 2.0 response with an `id` it does
+not remember sending a request for, it MUST ignore that response.
+
+> **Rationale** Clients, LSPs, and the network between them
+> can individually be unreliable, leading to clients that forget
+> `id`s they issued previously, or clients or LSPs thinking that
+> the other side may have restarted when it is the network
+> between them that failed.
+> Thus, random `id`s picked from a large space are the safest
+> when the client might lose any counter state (by crashing if
+> using an in-memory counter, or loss of persistent storage on
+> hardware without storage redundancy), and are resilient
+> against reconnections when both sides remained running.
+
+Clients MAY include additional information in their `id` for
+internal tracking, as long as the total `id` has sufficient
+entropy for universal uniqueness.
+
+Clients SHOULD internally impose a reasonable timeout, on
+the scale of minutes, for receiving a response for a request,
+and SHOULD treat a timeout event as a temporary server failure,
+and forget the `id` of the timed-out request.
+
+> **Rationale** The LSP might crash between the time the client
+> makes the request to the time it could complete processing of
+> the response, and thus lost track of the request.
+
+If the LSP is unable to deliver a response to the client due
+to a disconnection, it SHOULD treat this no differently from
+successfully delivering the response but the client then does
+not "follow up" on any action after that.
```

Comment:
See review comment above. If the client must ignore the response and forget about the request, the LSP shouldn't assume succesful delivery if the response does not arrive. OR the client should remember it sent the message.

Reply:
Always remember that a client can crash even after it receives the response.  The LSP being able to deliver a response 
 to the client is not a reliable knowledge that the client that has to do some CPU cycles actually currently has CPU cycles to do that and **then** proceed with whatever next step should be in the flow.  Again, consider the case where the LSP delivers the response successfully to the client, and while the client is about to send the next request in the flow, the user of the client is involved in a boating accident and loses their mobile device that is running the client.  So every LSP and every LSPS design has to be prepared for the client suddenly disappearing in the middle of a flow, and do something reasonable like time-bound things (a reason for why LSPS3 JIT Channels has a `valid_until` field) or have the client save state (a reason fro why LSPS3 JIT Channels has a `promise` field).

Reply:
There is a reason why BOLT specs have a `channel_reestablish`, there is a reason why channel acceptors are allowed to forget about a channel if it never confirms after 2 weeks (there is an edge case where the acceptor sends out the `funding_signed` successfully but the initiator crashes before it can persist the signature, the initiator thinks the flow got cancelled due to the disconnection and never broadcasts the channel funding, but the acceptor thinks the flow completed because it successfully sent `funding_signed`).  Any participant can crash at any time, and we should design our systems to be resilient against this.  Thus, these have to be part of the design of every LSPS specification.

Reply:
OK, will add to the section on Other LSPS Specifications.

Reply:
Done